### PR TITLE
[LLDB] Add a setting for printing ValueObject hex values without leading zeroes

### PR DESCRIPTION
**Pull request by walter-erquinigo**
As suggested by Greg in https://github.com/llvm/llvm-project/pull/66534, I'm adding a setting at the Target level that controls whether to show leading zeroes in hex ValueObject values.

This has the benefit of reducing the amount of characters displayed in certain interfaces, like VSCode.

### CODE REVIEW

**Review by clayborg — APPROVED** [review]
Nice! LGTM

File: lldb/source/Core/DumpDataExtractor.cpp [hunk 1]
```diff
@@ -620,10 +620,17 @@ lldb::offset_t lldb_private::DumpDataExtractor(
       case 2:
       case 4:
       case 8:
-        s->Printf(wantsuppercase ? "0x%*.*" PRIX64 : "0x%*.*" PRIx64,
-                  (int)(2 * item_byte_size), (int)(2 * item_byte_size),
-                  DE.GetMaxU64Bitfield(&offset, item_byte_size, item_bit_size,
-                                       item_bit_offset));
+        if (exe_scope->CalculateTarget()
```
Comment (clayborg) [inline lldb/source/Core/DumpDataExtractor.cpp:623]: Actually do you need to test if "exe_scope" is not NULL here?

**Review by clayborg — CHANGES_REQUESTED** [review]

**Comment by jimingham** [comment]
There are other places we print hex numbers that don't go through the DataExtractor.  For instance, the "location" for a ValueObject is printed directly, so this change wouldn't catch it.  It's a little off-putting if you set a global setting to "not print leading zeros" and then sometimes we still print leading zeros.

This is not an objection to the contents of this patch, but I think to be complete we're going to end up finding more places where we should be checking this policy setting.

**Comment by walter-erquinigo** [comment]
@jimingham , that's a good catch. Probably I should rename this because I want this setting to affect only SBValueObject::GetValue(), which is want ends up being sent to IDEs as a common interface. Trying to control all places that print as hex value is a bit too much and it'll be hard to enforce in the long term. Any particular suggestion? Because I can't really come up with a good name better than  `show-leading-zeroes-in-valueobject-values`.

**Comment by jimingham** [comment]
That really should be `show-leading-zeros-in-valueobject-variables-using-hex-format`, shouldn't it?  I think you can actually just document what it actually does in the help string and then you shouldn't have to make the name overly pedantic.   The one you have already is probably enough.

**Comment by clayborg** [comment]
> @jimingham , that's a good catch. Probably I should rename this because I want this setting to affect only SBValueObject::GetValue(), which is want ends up being sent to IDEs as a common interface. Trying to control all places that print as hex value is a bit too much and it'll be hard to enforce in the long term. Any particular suggestion? Because I can't really come up with a good name better than `show-leading-zeroes-in-valueobject-values`.

I think the setting name is ok as is. No one outside of LLDB developers have any idea what "valueobject" means. Also this setting is "show-hex-values-with-leading-zeroes" which does mention "values" so it can be ok for not all hex values emitted that are not values to have leading zeroes. The location is not a "value" technically and we might want some values coming out with leading zeroes like any command that dumps out table style output like "image dymp symtab" and other areas. Maybe we can name this setting:
```
show-hex-variable-values-with-leading-zeroes
```
The confirm this is for variable values only and not a system wide thing? 

If we switch the setting to be global in the target settings we can get the global target settings without needing to check the exe_scope. Thoughts?

**Comment by walter-erquinigo** [comment]
I like Greg's proposal of naming it `show-hex-variable-values-with-leading-zeroes`. Seems very clear. I also like the idea of making it a global target config, which would simplify the code.

**Comment by walter-erquinigo** [comment]
I've updated the PR following Greg's suggestions.

**Review by clayborg — COMMENTED** [review]
LGTM. Jim? You ok with this?

**Comment by jimingham** [comment]
Sure.

**Comment by walter-erquinigo** [comment]
Thanks Jim and Greg!

**Comment by DavidSpickett** [comment]
This is failing on Arm 32 bit (https://lab.llvm.org/buildbot/#/builders/17/builds/43300), I'll reproduce and fix it. I expect it just needs to expect 8 hex characters instead of 16.

**Comment by DavidSpickett** [comment]
https://github.com/llvm/llvm-project/commit/956860168b56f135b9facf3827ae9f3634d41f40

**Comment by walter-erquinigo** [comment]
Thanks, man!

**Pull request merged**